diff --git a/cmake-projects/jsonpath/CMakeLists.txt b/cmake-projects/jsonpath/CMakeLists.txt
index 3213379..f455318 100644
--- a/cmake-projects/jsonpath/CMakeLists.txt
+++ b/cmake-projects/jsonpath/CMakeLists.txt
@@ -19,26 +19,35 @@ IF(JSONC_FOUND)
 ENDIF()
 
 ADD_CUSTOM_COMMAND(
-  OUTPUT contrib/lemon
-  DEPENDS contrib/lemon.c contrib/lempar.c
-  COMMAND gcc -std=gnu17 -o contrib/lemon contrib/lemon.c
+  OUTPUT lemon
+  DEPENDS ${CMAKE_SOURCE_DIR}/contrib/lemon.c ${CMAKE_SOURCE_DIR}/contrib/lempar.c
+  COMMAND gcc -std=gnu17 -o lemon -I "${CMAKE_SOURCE_DIR}" "${CMAKE_SOURCE_DIR}/contrib/lemon.c"
   COMMENT "Generating lemon parser generator"
 )
 
 ADD_CUSTOM_COMMAND(
-  OUTPUT parser.c
-  DEPENDS parser.y contrib/lemon
-  COMMAND ./contrib/lemon parser.y
+  OUTPUT ${CMAKE_SOURCE_DIR}/parser.c
+  DEPENDS ${CMAKE_SOURCE_DIR}/parser.y lemon
+  COMMAND cd "${CMAKE_SOURCE_DIR}" && ${CMAKE_BINARY_DIR}/lemon -T${CMAKE_SOURCE_DIR}/contrib/lempar.c parser.y
   COMMENT "Generating parser.c"
 )
 
 FIND_PATH(ubox_include_dir libubox/list.h)
-INCLUDE_DIRECTORIES(${ubox_include_dir})
+IF(ubox_include_dir MATCHES "NOTFOUND$")
+  ADD_SUBDIRECTORY(libubox)
+  INCLUDE_DIRECTORIES(libubox)
+ELSE()
+  INCLUDE_DIRECTORIES(${ubox_include_dir})
+ENDIF()
 
 SET_PROPERTY(DIRECTORY APPEND PROPERTY ADDITIONAL_MAKE_CLEAN_FILES "lemon;parser.h;parser.out")
 SET_SOURCE_FILES_PROPERTIES("parser.c" PROPERTIES GENERATED TRUE)
+
+SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
+SET(BUILD_SHARED_LIBS OFF)
+
 ADD_EXECUTABLE(jsonpath main.c ast.c lexer.c parser.c matcher.c)
 find_library(json NAMES json-c)
-TARGET_LINK_LIBRARIES(jsonpath ubox ${json})
+TARGET_LINK_LIBRARIES(jsonpath ubox-static ${json})
 
 INSTALL(TARGETS jsonpath RUNTIME DESTINATION bin)
diff --git a/cmake-projects/jsonpath/ast.c b/cmake-projects/jsonpath/ast.c
index 3f88d94..9a9b37a 100644
--- a/cmake-projects/jsonpath/ast.c
+++ b/cmake-projects/jsonpath/ast.c
@@ -21,7 +21,7 @@
 #include <stdlib.h>
 #include <stdarg.h>
 #include <string.h>
-#include <libubox/utils.h>
+#include "libubox/utils.h"
 
 struct jp_opcode *
 jp_alloc_op(struct jp_state *s, int type, int num, char *str, ...)
diff --git a/cmake-projects/jsonpath/main.c b/cmake-projects/jsonpath/main.c
index 5758ab8..f468b02 100644
--- a/cmake-projects/jsonpath/main.c
+++ b/cmake-projects/jsonpath/main.c
@@ -26,7 +26,7 @@
 	#include <json-c/json.h>
 #endif
 
-#include <libubox/list.h>
+#include "libubox/list.h"
 
 #include "lexer.h"
 #include "parser.h"
