diff --git a/cmake-projects/jsonpath/libubox/CMakeLists.txt b/cmake-projects/jsonpath/libubox/CMakeLists.txt
index 1a1b7d5..f4b3675 100644
--- a/cmake-projects/jsonpath/libubox/CMakeLists.txt
+++ b/cmake-projects/jsonpath/libubox/CMakeLists.txt
@@ -37,6 +37,7 @@ IF(NOT HAVE_GETTIME)
 ENDIF()
 
 FILE(GLOB headers *.h)
+IF(OFF)
 INSTALL(FILES ${headers}
 	DESTINATION include/libubox
 )
@@ -44,6 +45,7 @@ INSTALL(TARGETS ubox ubox-static
 	ARCHIVE DESTINATION lib
 	LIBRARY DESTINATION lib
 )
+ENDIF()
 
 ADD_SUBDIRECTORY(lua)
 ADD_SUBDIRECTORY(examples)
@@ -80,6 +82,7 @@ IF(EXISTS ${json})
 	ADD_LIBRARY(json_script SHARED json_script.c)
 	TARGET_LINK_LIBRARIES(json_script ubox)
 
+	IF(OFF)
 	INSTALL(TARGETS blobmsg_json blobmsg_json-static jshn json_script
 		ARCHIVE DESTINATION lib
 		LIBRARY DESTINATION lib
@@ -90,6 +93,7 @@ IF(EXISTS ${json})
 	INSTALL(FILES ${scripts}
 		DESTINATION share/libubox
 	)
+	ENDIF()
 
 ENDIF()
 
diff --git a/cmake-projects/jsonpath/libubox/assert.h b/cmake-projects/jsonpath/libubox/assert.h
deleted file mode 100644
index 84f5471..0000000
--- a/cmake-projects/jsonpath/libubox/assert.h
+++ /dev/null
@@ -1,9 +0,0 @@
-#pragma once
-
-#ifdef NDEBUG
-#undef NDEBUG
-#include <assert.h>
-#define NDEBUG
-#else
-#include <assert.h>
-#endif
diff --git a/cmake-projects/jsonpath/libubox/avl.c b/cmake-projects/jsonpath/libubox/avl.c
index 79ea5c7..d1d94ee 100644
--- a/cmake-projects/jsonpath/libubox/avl.c
+++ b/cmake-projects/jsonpath/libubox/avl.c
@@ -45,7 +45,7 @@
 #include <string.h>
 
 #include "avl.h"
-#include "assert.h"
+#include "ubox_assert.h"
 #include "list.h"
 
 /**
diff --git a/cmake-projects/jsonpath/libubox/base64.c b/cmake-projects/jsonpath/libubox/base64.c
index 1bf2177..bdf4edd 100644
--- a/cmake-projects/jsonpath/libubox/base64.c
+++ b/cmake-projects/jsonpath/libubox/base64.c
@@ -66,7 +66,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include "assert.h"
+#include "ubox_assert.h"
 #include "utils.h"
 
 static const char Base64[] =
diff --git a/cmake-projects/jsonpath/libubox/ubox_assert.h b/cmake-projects/jsonpath/libubox/ubox_assert.h
new file mode 100644
index 0000000..84f5471
--- /dev/null
+++ b/cmake-projects/jsonpath/libubox/ubox_assert.h
@@ -0,0 +1,9 @@
+#pragma once
+
+#ifdef NDEBUG
+#undef NDEBUG
+#include <assert.h>
+#define NDEBUG
+#else
+#include <assert.h>
+#endif
