diff -ru orig.jsonpath/libubox/CMakeLists.txt jsonpath/libubox/CMakeLists.txt
--- orig.jsonpath/libubox/CMakeLists.txt	2018-10-10 15:36:15.797999061 -0300
+++ jsonpath/libubox/CMakeLists.txt	2018-10-10 15:21:49.998976072 -0300
@@ -31,6 +31,7 @@
 ENDIF()
 
 FILE(GLOB headers *.h)
+IF(OFF)
 INSTALL(FILES ${headers}
 	DESTINATION include/libubox
 )
@@ -38,6 +39,7 @@
 	ARCHIVE DESTINATION lib
 	LIBRARY DESTINATION lib
 )
+ENDIF()
 
 ADD_SUBDIRECTORY(lua)
 ADD_SUBDIRECTORY(examples)
@@ -57,6 +59,7 @@
 	ADD_LIBRARY(json_script SHARED json_script.c)
 	TARGET_LINK_LIBRARIES(json_script ubox)
 
+	IF(OFF)
 	INSTALL(TARGETS blobmsg_json blobmsg_json-static jshn json_script
 		ARCHIVE DESTINATION lib
 		LIBRARY DESTINATION lib
@@ -67,5 +70,6 @@
 	INSTALL(FILES ${scripts}
 		DESTINATION share/libubox
 	)
+	ENDIF()
 
 ENDIF()
